tox.ini

```diff
@@ -4,11 +4,115 @@
 # and then run "tox" from this directory.
 
 [tox]
-envlist = flake8, py27, py33, py34, py35
+envlist =
+    py27,
+    py34,
+    py35,
+    py36,
+    docs,
+    isort,
+    flake8
+    coverage-report,
 
 [testenv]
-commands = python setup.py test
+usedevelop = True
+extras =
+    test
+
+commands =
+    python setup.py test
+    mkdir -p {toxinidir}/reports/coverage {toxinidir}/reports/pytest
+    pytest --cov=lib --cov-report=xml --html={toxinidir}/reports/pytest/report-{envname}.html --self-contained-html lib/diazo/tests {posargs}
+
+setenv =
+    COVERAGE_FILE=.coverage.{envname}
+
+deps =
+    pdbpp
+    pytest
+    pytest-cov
+    pytest-remove-stale-bytecode
+    pytest-mock
+    pytest-html
+
+whitelist_externals =
+    mkdir
+
+[testenv:coverage-report]
+basepython = python2.7
+deps = coverage
+
+setenv =
+    COVERAGE_FILE=.coverage
+
+skip_install = true
+
+commands =
+    coverage erase
+    coverage combine
+    coverage report
+    coverage html
+    coverage xml
+
+[testenv:isort]
+basepython = python2.7
+deps = isort
+commands =
+    #isort --check-only --recursive {toxinidir}/lib
+    isort --check-only --recursive {toxinidir}/lib
+
+[testenv:isort-apply]
+basepython = python2.7
+deps = isort
+commands =
+    isort --apply --recursive {toxinidir}/lib
 
 [testenv:flake8]
-deps = flake8
-commands = flake8 lib setup.py
+basepython = python2.7
+deps =
+    flake8
+    # helper to generate HTML reports:
+    flake8-html
+    flake8-blind-except
+    flake8-coding
+    flake8-commas
+    flake8-debugger
+    flake8-deprecated
+    flake8-isort
+    flake8-pep3101
+#    flake8-plone-api
+#    flake8-plone-hasattr
```

Comment:
Actually I should have renamed this to flake8-hasattr as it is no plone specific, but a generic python one: hasattr swallows exceptions and that is usually what you don't want, usually of course.